UKHomeOffice/sgmr-service
# SPCR-534 Fix optional skipper postcode & hide submit/save buttons on submitted voyages

### AC / Description of changes
- fix the skipper postcode so it is optional to user, but still passes a string as required to API
- hide submit button if voyage is in submitted, presubmitted, or failed state
- hide save draft button if voyage is in submitted, presubmitted, or failed state


### To Test

- go to voyage skipper page
- submit page with no postcode
- > page should not error
- go back
- submit page with postcode
- > page should not error
- > postcode should be visible on CYA page


- go to a draft report
- > CYA page should have all 3 buttons
- submit (green)
- cancel (red)
- save as draft (grey)

- go to a submitted report
- > CYA page should have only cancel button
- go to a cancelled report
- > CYA page should have no buttons


### Notes

---

**Remember to merge to feature branches not `master`.**
